### PR TITLE
Added new test

### PR DESCRIPTION
**Pull request by acul021**
Hi, I added a new test

### CODE REVIEW

File: src/test/java/edu/kit/informatik/informatik/NetworkTestByAcul021.java [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+            net.getRoute(ip("85.193.148.81"), ip("122.117.67.158")));
+        assertEquals(ip("122.117.67.158", "141.255.1.133", "85.193.148.81", "231.189.0.127", "39.20.222.120"),
+            net.getRoute(ip("122.117.67.158"), ip("39.20.222.120")));
+
+        assertTrue(() -> net.add(net("(0.0.0.0 1.1.1.1 2.2.2.2 (3.3.3.3 4.4.4.4))")));
+        assertEquals(List.of(), net.getRoute(ip("39.20.222.120"), ip("4.4.4.4")));
+    }
+
+    @Test
+    public void testEqualsAndHashCode() throws ParseException {
```
Comment (lscobe16) [inline src/test/java/edu/kit/informatik/informatik/NetworkTestByAcul021.java:239]: The task description does not require us to implement equals and hashCode on Network.

File: src/test/java/edu/kit/informatik/informatik/NetworkTestByAcul021.java [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+        assertThrows(RuntimeException.class, () -> new Network(root, invalidIPs));
+        assertThrows(RuntimeException.class, () -> new Network(root, ip("0.0.0.0", "0.0.0.0")));
+
+        Network testNetwork = new Network(root, IPs);
+        IPs.clear();
+        assertNotEquals(0, testNetwork.list().size());
+    }
+
+    @Test
+    public void testParsingConstructor() throws ParseException {
```
Comment (lscobe16) [inline src/test/java/edu/kit/informatik/informatik/NetworkTestByAcul021.java:64]: Use of a ParameterizedTest would be nice and would give more granular test results

**Review by lscobe16 — COMMENTED** [review]
Hey, thanks for sharing your tests. They are pretty comprehensive!
I made some annotations

File: src/test/java/edu/kit/informatik/informatik/NetworkTestByAcul021.java (hunk 1 above)
Comment (koerihub) [reply src/test/java/edu/kit/informatik/informatik/NetworkTestByAcul021.java:239]: `equals` is required, and according to `hashCode`s contract two equal objects must produce the same hash, so this is a reasonable test case.

**Comment by koerihub** [comment]
Merged. Additional improvements like splitting the test cases into different classes and using parameterized tests may be done in separate merge requests.

**Pull request closed without merging**